app.ts

```diff
@@ -0,0 +1,286 @@
+
+	// Clears Table
+	if (content !== null) {
+		content.innerHTML = "";
+	}
+
+	// Displays The Current Results Being Shown
+	let currentStats = `Showing results from ${fromParameter} to ${toParameter} out of ${recordCount} results.`;
+	if (pageStats !== null) {
+		pageStats.innerHTML = currentStats;
+	}
+
+	return fetch(`http://localhost:2050/records?from=${fromParameter}&to=${toParameter}`, {
+		method: "GET",
+		headers: { "Content-Type": "application/json" },
+	})
+		.then(handleErrors)
+		.then((res: Response) => res.json())
+		.then((contentData: string | string[]) => {
+			for (let x of contentData) {
+				appendTableContent(x);
+			}
+		})
+		.catch((error: Error) => {
+			console.log(error);
+		});
+}
+
+// Gets Total Of All Records
+function getRecordCount(): Promise<void> {
+	return fetch("http://localhost:2050/recordCount", {
+		method: "GET",
+		headers: { "Content-Type": "application/json" },
+	})
+		.then(handleErrors)
+		.then((res: Response) => res.json())
+		.then((data: number) => {
+			recordCount = data;
+		})
+		.catch((error: Error) => {
+			console.log(error);
+		});
+}
+
+//// Debounce
+function debounce(fn: () => void, delay: number) {
+	let timer: number
+	return function () {
+		clearTimeout(timer);
+		timer = setTimeout(() => {
+			fn();
+		}, delay);
+	};
+}
+
+//// Sizing And Resizing
+function resizing() {
+	let end = fromParameter + getNoOfRows();
+	let toParameter: number;
+	let maxRecordsID = recordCount - 1;
+
+	if (end > maxRecordsID) {
+		toParameter = maxRecordsID;
+		fromParameter = toParameter - getNoOfRows();
+	}
+	getTable();
+}
+
+//// Navigation
+// Next
+class Next {
+	nextButton: HTMLElement | null;
+	nextTimer: number = 0
+
+	constructor() {
+		this.nextButton = document.getElementById("next");
+
+		const nextDebounce = (fn: () => void, delay: number) => {
+			return () => {
+				clearTimeout(this.nextTimer);
+				this.nextTimer = setTimeout(() => {
+					fn();
+				}, delay);
+			};
+		};
+
+		let next = () => {
+			let toParameter = getParameters(fromParameter);
+			let maxRecordsID = recordCount - 1;
+
+			if (toParameter === maxRecordsID) {
+				alert("You have reached the final page");
+			}
+
+			let nextAmount = toParameter - fromParameter + 1;
+			fromParameter = fromParameter + nextAmount;
+			toParameter = fromParameter + getNoOfRows();
+
+			let end = fromParameter + getNoOfRows();
+
+			if (end > maxRecordsID) {
+				toParameter = maxRecordsID;
+				fromParameter = toParameter - getNoOfRows();
+			}
+
+			nextDebounce(getTable, 500)()
+		};
+
+		if (this.nextButton) {
+			this.nextButton.addEventListener("click", next);
+		}
+	}
+}
+
+// Previous
+class Prev {
+	prevButton: HTMLElement | null;
+	prevTimer: number = 0
+
+	constructor() {
+		this.prevButton = document.getElementById("prev");
+
+		const prevDebounce = (fn: () => void, delay: number) => {
+			return () => {
+				clearTimeout(this.prevTimer);
+				this.prevTimer = setTimeout(() => {
+					fn();
+				}, delay);
+			};
+		};
```

Comment:
I think you should rather change this to not return a function, that way you don't have to have that weird `()` on line 231.